runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -3334,11 +3332,16 @@ private void onKafkaReset(
             Flyweight mqttResetEx = EMPTY_OCTETS;
             if (error != -1)
             {
-                mqttResetEx = mqttSessionResetExRW.wrap(sessionExtBuffer, 0, sessionExtBuffer.capacity())
+                MqttResetExFW.Builder mqttResetBuilder =
+                    mqttSessionResetExRW.wrap(sessionExtBuffer, 0, sessionExtBuffer.capacity())
                     .typeId(mqttTypeId)
-                    .reasonCode(MQTT_REASON_CODES.get(kafkaResetEx.error()) & 0xff)
-                    .reason(MQTT_REASONS.getOrDefault(kafkaResetEx.error(), EMPTY_STRING))
-                    .build();
+                    .reasonCode(MQTT_REASON_CODES.get(kafkaResetEx.error()));
+                final String16FW reason = MQTT_REASONS.get(kafkaResetEx.error());
+                if (reason != null)
+                {
+                    mqttResetBuilder.reason(reason);
+                }
+                mqttResetEx = mqttResetBuilder.build();
```

Comment:
We already have `error` above, so no need to call `kafkaResetEx.error()` again multiple times.
I think it's okay to pass `null` to `.reason(...)` on the builder, so probably no need to split the builder and make the code more complicated.

We likely need just a very small diff from the previous code.